src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/NFloatTests.cs

```diff
@@ -1606,6 +1606,165 @@ public static void Root32(float x, int n, float expectedResult, float allowedVar
             AssertExtensions.Equal(expectedResult, NFloat.Root(x, n), allowedVariance);
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.Is32BitProcess))]
+        [InlineData( float.NaN,    float.NaN,    0.0f)]
+        [InlineData( 1.0f,         0.0f,         0.0f)]
+        [InlineData( 0.540302306f, 0.318309886f, CrossPlatformMachineEpsilon32)]
+        [InlineData( 0.204957194f, 0.434294482f, CrossPlatformMachineEpsilon32)]
+        [InlineData( 0.0f,         0.5f,         0.0f)]
+        [InlineData(-0.416146837f, 0.636619772f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.570233249f, 0.693147181f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.605699867f, 0.707106781f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.781211892f, 0.785398163f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-1.0f,         1.0f,         0.0f)]
+        [InlineData(-0.919764995f, 0.871620833f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.266255342f, 0.585786438f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.179057946f, 0.557304959f, CrossPlatformMachineEpsilon32)]
+        [InlineData( 0.220584041f, 0.429203673f, CrossPlatformMachineEpsilon32)]
+        [InlineData( 0.581195664f, 0.302585093f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.633255651f, 0.718281828f, CrossPlatformMachineEpsilon32)]
+        [InlineData(-0.902685362f, 0.858407346f, CrossPlatformMachineEpsilon32)]
```

Comment:
float test cases seem to be duplicated for `SingleTests.cs`, `HalfTests.cs` and `NFloatTests.cs` (for 32 bit systems). Have you considered having single static factory method for reusing it for all of them? Or having a single test for all of them?

Example:

```cs
[Fact]
public static void AcosPiTest(float value, float expectedResult, float allowedVariance)
{
    AssertExtensions.Equal(expectedResult, float.AcosPi(value), allowedVariance);
    
    AssertExtensions.Equal((Half)expectedResult, Half.AcosPi((Half)value), (Half)allowedVariance);
    
    if (PlatformDetection.Is32BitProcess)
        AssertExtensions.Equal(expectedResult, (float)NFloat.AcosPi(value), allowedVariance);
}
```

Reply:
I don't think its worth sharing. The values can/should be duplicated out as they may need to differ or diverge.

In the case of `Half`, if we ever get literal support we'd change it all.